lib/node/nodes/kmeans.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var Node = require('../node');
+var dot = require('dot');
+
+var TYPE = 'kmeans';
+
+var PARAMS = {
+    source : Node.PARAM.NODE(Node.GEOMETRY.POINT),
+    clusters : Node.PARAM.NUMBER
+};
+
+var KMeans  = Node.create(TYPE, PARAMS, {cache: false});
```

Comment:
This is really important, all the analysis that need knoledge about the full dataset should:

- set cached: true
- rethinking about them because we should try to avoid them (they are not easily parallelizable)